### PR TITLE
python3Packages.pyjwt: disable failing test for 1.7.1

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
With the update to > 2.x a couple of packages need to override PyJWT. In #121891 the check part was updated.

```bash
[...]
builder for '/nix/store/8g424yx2j95dm4x7ivl32f4pzljbbns8-python3.8-pyjwt-1.7.1.drv' failed with exit code 1; last 10 log lines:
  tests/test_exceptions.py                 4      0   100%
  tests/test_jwt.py                        8      0   100%
  tests/test_utils.py                     16      0   100%
  tests/utils.py                          21     11    48%   20-34
  ------------------------------------------------------------------
  TOTAL                                 2026    195    90%
  
  =========================== short test summary info ============================
  FAILED tests/contrib/test_algorithms.py::TestEcdsaAlgorithms::test_ec_verify_should_return_false_if_signature_invalid
  ====== 1 failed, 178 passed, 8 skipped, 1 xfailed, 140 warnings in 1.19s =======
```

The failing test passes if PyJWT is > 2.0 but not on 1.7.1.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122380` at 2ee56622 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>24 packages marked as broken and skipped:</summary>
  <ul>
    <li>bareos</li>
    <li>python38Packages.arviz</li>
    <li>python38Packages.dm-sonnet</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.ignite</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.pymc3</li>
    <li>python38Packages.python-smarttub</li>
    <li>python38Packages.pytorchWithCuda</li>
    <li>python38Packages.rl-coach</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/1c320e201cbb03cd235e0ae666d99769">python38Packages.djangorestframework-jwt</a></li>
    <li><a href="https://gist.github.com/683c212a194ab5e1d6c9a54072805ccf">python38Packages.flask-appbuilder</a></li>
    <li><a href="https://gist.github.com/8ff160b82e08fedfa63018a7ef371e59">python38Packages.globus-sdk</a></li>
    <li><a href="https://gist.github.com/a07a9a1c4317b5c85f9ddca15d314ff7">python38Packages.twilio</a></li>
    <li><a href="https://gist.github.com/9966e00f660e143a8903e2b9300eb467">python39Packages.djangorestframework-jwt</a></li>
    <li><a href="https://gist.github.com/cb4ec7002cf29bf22bb11117f23371fb">python39Packages.flask-appbuilder</a></li>
    <li><a href="https://gist.github.com/dfe2e54f2bb80508ba417176f7100451">python39Packages.globus-sdk</a></li>
    <li><a href="https://gist.github.com/5a14c419350cda727a18ecdb853654b3">python39Packages.twilio</a></li>
  </ul>
</details>
<details>
  <summary>339 packages skipped due to time constraints:</summary>
  <ul>
    <li>apache-airflow</li>
    <li>appdaemon</li>
    <li>apprise (python38Packages.apprise)</li>
    <li>azure-cli</li>
    <li>beets</li>
    <li>beetsExternalPlugins.alternatives</li>
    <li>beetsExternalPlugins.copyartifacts</li>
    <li>beetsExternalPlugins.extrafiles</li>
    <li>breezy (python38Packages.breezy)</li>
    <li>cabal2nix</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>89 packages built successfully:</summary>
  <ul>
    <li>alerta-server</li>
    <li>buildbot (python38Packages.buildbot)</li>
    <li>buildbot-full (python38Packages.buildbot-full)</li>
    <li>buildbot-ui (python38Packages.buildbot-ui)</li>
    <li>cligh</li>
    <li>cloud-init</li>
    <li>eolie</li>
    <li>mathlibtools (python38Packages.mathlibtools)</li>
    <li>matrix-synapse</li>
    <li>pleroma-bot</li>
    <li>python38Packages.PyGithub</li>
    <li>python38Packages.adal</li>
    <li>python38Packages.atlassian-python-api</li>
    <li>python38Packages.azure-core</li>
    <li>python38Packages.azure-datalake-store</li>
    <li>python38Packages.azure-identity</li>
    <li>python38Packages.azure-storage-blob</li>
    <li>python38Packages.criticality-score</li>
    <li>python38Packages.discogs_client</li>
    <li>python38Packages.django-oauth-toolkit</li>
    <li>python38Packages.djangorestframework-simplejwt</li>
    <li>python38Packages.exchangelib</li>
    <li>python38Packages.flask-jwt-extended</li>
    <li>python38Packages.gidgethub</li>
    <li>python38Packages.google-auth-oauthlib</li>
    <li>python38Packages.homepluscontrol</li>
    <li>python38Packages.jupyterhub</li>
    <li>python38Packages.keep</li>
    <li>python38Packages.kubernetes</li>
    <li>python38Packages.lazr-restfulclient</li>
    <li>python38Packages.msal</li>
    <li>python38Packages.msal-extensions</li>
    <li>python38Packages.msrest</li>
    <li>python38Packages.msrestazure</li>
    <li>python38Packages.mwoauth</li>
    <li>python38Packages.oauthenticator</li>
    <li>python38Packages.oauthlib</li>
    <li>python38Packages.ondilo</li>
    <li>python38Packages.pyatmo</li>
    <li>python38Packages.pyflume</li>
    <li>python38Packages.pyfxa</li>
    <li>python38Packages.pyjwt</li>
    <li>python38Packages.requests_oauthlib</li>
    <li>python38Packages.ring-doorbell</li>
    <li>python38Packages.snowflake-connector-python</li>
    <li>python38Packages.snowflake-sqlalchemy</li>
    <li>python38Packages.webexteamssdk</li>
    <li>python39Packages.PyGithub</li>
    <li>python39Packages.adal</li>
    <li>python39Packages.atlassian-python-api</li>
    <li>python39Packages.azure-core</li>
    <li>python39Packages.azure-datalake-store</li>
    <li>python39Packages.azure-identity</li>
    <li>python39Packages.azure-storage-blob</li>
    <li>python39Packages.buildbot</li>
    <li>python39Packages.buildbot-full</li>
    <li>python39Packages.buildbot-ui</li>
    <li>python39Packages.criticality-score</li>
    <li>python39Packages.discogs_client</li>
    <li>python39Packages.django-oauth-toolkit</li>
    <li>python39Packages.djangorestframework-simplejwt</li>
    <li>python39Packages.exchangelib</li>
    <li>python39Packages.flask-jwt-extended</li>
    <li>python39Packages.gidgethub</li>
    <li>python39Packages.google-auth-oauthlib</li>
    <li>python39Packages.homepluscontrol</li>
    <li>python39Packages.jupyterhub</li>
    <li>python39Packages.keep</li>
    <li>python39Packages.kubernetes</li>
    <li>python39Packages.lazr-restfulclient</li>
    <li>python39Packages.mathlibtools</li>
    <li>python39Packages.msal</li>
    <li>python39Packages.msal-extensions</li>
    <li>python39Packages.msrest</li>
    <li>python39Packages.msrestazure</li>
    <li>python39Packages.mwoauth</li>
    <li>python39Packages.oauthenticator</li>
    <li>python39Packages.oauthlib</li>
    <li>python39Packages.ondilo</li>
    <li>python39Packages.pyatmo</li>
    <li>python39Packages.pyflume</li>
    <li>python39Packages.pyfxa</li>
    <li>python39Packages.pyjwt</li>
    <li>python39Packages.requests_oauthlib</li>
    <li>python39Packages.ring-doorbell</li>
    <li>python39Packages.snowflake-connector-python</li>
    <li>python39Packages.snowflake-sqlalchemy</li>
    <li>python39Packages.webexteamssdk</li>
    <li>sourcehut.dispatchsrht</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 122380` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>home-assistant</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyjwt/default.nix [hunk 1]
```diff
@@ -29,6 +29,11 @@ buildPythonPackage rec {
     pytestCheckHook
   ];
 
+  disabledTests = lib.optionals (lib.versionOlder version "2.0") [
+    # ecdsa changed internal behavior, required if one does 1.7.1 overriding
+    "ec_verify_should_return_false_if_signature_invalid"
```
Comment (mweinelt) [inline pkgs/development/python-modules/pyjwt/default.nix:34]: And after merging I noticed the test name is wrong.

```
FAILED tests/contrib/test_algorithms.py::TestEcdsaAlgorithms::test_ec_verify_should_return_false_if_signature_invalid
```

Comment (mweinelt) [reply pkgs/development/python-modules/pyjwt/default.nix:34]: And even after patching this home-assistant doesn't build.

```diff
diff --git a/pkgs/development/python-modules/pyjwt/default.nix b/pkgs/development/python-modules/pyjwt/default.nix
index 104dadbd396..b2e642dd5cf 100644
--- a/pkgs/development/python-modules/pyjwt/default.nix
+++ b/pkgs/development/python-modules/pyjwt/default.nix
@@ -31,7 +31,7 @@ buildPythonPackage rec {
 
   disabledTests = lib.optionals (lib.versionOlder version "2.0") [
     # ecdsa changed internal behavior, required if one does 1.7.1 overriding
-    "ec_verify_should_return_false_if_signature_invalid"
+    "test_ec_verify_should_return_false_if_signature_invalid"
   ];
 
   pythonImportsCheck = [ "jwt" ];
```

```
=========================== short test summary info ============================
FAILED tests/contrib/test_algorithms.py::TestEcdsaAlgorithms::test_ec_verify_should_return_false_if_signature_invalid
====== 1 failed, 178 passed, 8 skipped, 1 xfailed, 140 warnings in 2.02s =======
builder for '/nix/store/q8jhyxiksz54n85sbnrbr05n6ggbxqd1-python3.8-pyjwt-1.7.1.drv' failed with exit code 1
```

I'll have to go shopping now and will check back later.

**Pull request merged**